appeals/web/src/server/app/components/file-downloader.component.js

```diff
@@ -60,4 +60,61 @@ const getDocumentDownload = async ({ apiClient, params, session }, response) =>
 	return response.status(200);
 };
 
-export default getDocumentDownload;
+/**
+ * Download one document or redirects to its url if preview is active
+ *
+ * @param {{apiClient: import('got').Got, params: {caseId: number, guid: string, preview?: string, version: string}, session: SessionWithAuth}} request
+ * @param {Response} response
+ * @returns {Promise<Response>}
+ */
+export const getDocumentDownloadByVersion = async ({ apiClient, params, session }, response) => {
+	const { guid: fileGuid, preview, caseId, version } = params;
+
+	const filesInfo = await apiClient.get(`appeals/${caseId}/documents/${fileGuid}/versions`).json();
+	const fileToDownloadInfo = filesInfo.documentVersion.find(
+		(/** @type {{ version: { toString: () => string; }; }} */ fileInfo) =>
```

Comment:
It seems likely this type will be reused, could it be updated to a named type rather than being defined inline?

Reply:
No longer applies